feat: Security and compliance reporting TAROT-2591 TAROT-2652 (#2052)
codacy/docs: docs/organizations/managing-security-and-risk.md

> # Managing security and risk

Reviewer: regarding use cases, I'd highlight:

1. Understand if the current state of security is good or bad.
2. Understand how security posture progresses over time.
3. Find and prioritize the most problematic areas or vulnerabilities that need attention.
4. Report status and progress to your stakeholders.

this presentation has more details about which section meets each use case: https://docs.google.com/presentation/d/1qyPTbfVPZJs7nqKRPm2H79Gl54bOpcHSAsJEJ5x1fu0/edit#slide=id.g2c087d3ee8e_0_42

Author: Thanks! Tweaked with more info on use cases ([commit](https://github.com/codacy/docs/pull/2052/commits/f3ead77a43017177c1f8ca8fdf835289aa782c29), [commit](https://github.com/codacy/docs/pull/2052/commits/3f60b6e6479645ef00dee3f5976327d822694f2b))

-   [Overview](#dashboard)
-   [Findings](#item-list)

In addition, on these pages, you can [share filtered views of findings](#sharing-filtered-view), [export findings as a CSV file](#exporting-the-security-item-list), and [review severity rules and integration settings](#reviewing-settings)

## Overview {: id="dashboard"}

The **Security and risk management overview** page provides a high-level view of the security posture of your organization, including the number of open findings, the distribution of open findings by severity, the history of finding resolution, and a breakdown of the most high-risk repositories and most detected security categories.

Use this page to assess your organization's security posture and its progress over time, identify areas for improvement, and share findings with stakeholders.

To access the overview page, select an organization from the top navigation bar and select **Security and risk** on the left navigation sidebar.

![Security and risk management overview page](images/security-risk-management-overview.png)

The overview page includes six panels:

-   [Open findings overview](#open-findings-overview)

> -   [open findings distribution](#open-findings-distribution)

Reviewer 2: Small tweak:

```suggestion
-   [Open findings distribution](#open-findings-distribution)
```

> -   [open findings history](#open-findings-history)

Reviewer 2: Small tweak:

```suggestion
-   [Open findings history](#open-findings-history)
```

### Top 10 common security categories

The **Top 10 common security categories** list shows the most common security categories of open findings, ordered by count.

To access the findings page with the corresponding filter applied, click on a category.

![Security and risk management top categories panel](images/security-risk-management-overview-top-categories.png)

## Findings {: id="item-list"}

The **Security and risk management findings** page displays a filtered list of findings. By default, this list is sorted by status, and you can click the **First detected** column name to sort the findings by the detection date. Use this page to review and prioritize findings and track the progress of your security efforts.

To access the findings page, access the [overview page](#dashboard) and click the **Findings** tab.

![Security and risk management findings page](images/security-risk-management-findings.png)

When viewing the findings, you can update the filtering criteria by clicking the **Severity**, **Status**, **Repository**, or **Security category** drop-downs above the list.

> To find out more about a finding in the list, click its **Details** column to navigate to the finding details on the source platform. Click **View all findings** to get back to the findings list.

Reviewer 2: @nicklem I'm not sure how this works when the source platform is not Codacy... If the source is not Codacy, maybe the **View all findings** link doesn't apply, right?

**Side note:** Are we still presenting findings from other sources? Why don't we have that information on the list anymore, and we can't filter by source? 🤔

Reviewer 2: Removing the reference to **View all findings** link, as it doesn't apply to all sources (and it's not really needed, the UX is self-explanatory).

```suggestion
To find out more about a finding in the list, click its **Details** column to navigate to the finding details on the source platform.
```

![Security and risk management finding details](images/security-risk-management-finding-details.png)

## Sharing a filtered view of findings {: id="sharing-filtered-view"}

To share the current view of the overview or findings page, click the **Copy URL** button in the top right-hand corner of the page. This action copies the URL with the current filters applied to the clipboard.

## Exporting findings {: id="exporting-the-security-item-list"}

!!! info "This feature is available only to organization admins and organization managers"

To export a list of findings as a CSV file, click the options menu in the top right-hand corner of the page and select **Export findings (.csv)**. The exported list always includes all findings, ignoring any applied filters.

> ## Reviewing severity rules and integration settings {: id="reviewing-settings"}

Reviewer 2: @nicklem I find this section a bit confusing...

1. **Review** the severity rules gives me the expectation that I can update the defined rules somehow. However, it doesn't seem possible, but I don't have any information about it in this section, or in [the linked section](https://github.com/codacy/docs/pull/2052/files#diff-30a735ab496ba1650575435a2195b32444487dbb269bb4b685b2e1ad8c7ebe53R202).
2. For the Jira integration settings, the link takes me to the integration installation docs, where I don't find any details about the integration settings.
3. Also, when I click the link **View integrations** on the **Security and risk** screen menu, I'm redirected to the Git provider integration settings... (more confused!)

Author: @claudiacarpinteiro I [tweaked](https://github.com/codacy/docs/pull/2052/commits/22656af7a6f315e59e8677deb01ccd2dab2a01f5) to address your points.

It makes sense to update the CTA on the button from "View integrations" to "Manage Jura integration", since it's the only third-party SRM integration for now.

<img width="238" alt="image" src="https://github.com/codacy/docs/assets/11616750/523ba413-52dc-4602-86a0-8bc97c7801e7">

Thoughts?

cc @joanasteodoro @alerizzo

Reviewer 3: don't forget about slack! 🙌 that's currently a "security-only" integration right now

Author: Fair enough, although there's a mismatch with the context the user is in (the SRM page). The integration does push data to Slack, but has no immediate effect on the SRM dashboards.

Let's stick with the current CTA, "View integrations". I'll revert my suggestion.

# Managing security and risk

The Security and risk management feature helps you <span class="skip-vale">quickly</span> identify, track, and address security across your organization by automatically opening time-bound, prioritized findings whenever security problems are detected in your organization repositories, in your [connected Jira instance](./integrations/jira-integration.md), or during [penetration testing](https://go.codacy.com/pen-testing-product)

Under Security and risk management, you can find the following pages to help you monitor security:

-   [Overview](#dashboard)
-   [Findings](#item-list)

> 

Reviewer 2: IMO, this approach of having a "table of contents" in the introduction for the main areas only (and not including all the sections) it's much cleaner and reads better. 👍 
Though I was expecting here that sentence you intended to add, mentioning the possible actions (sharing and exporting). I believe it would fit great.

-   [Top 10 common security categories](#top-10-common-security-categories)

To limit the information displayed in each panel to a specific set of repositories, use the filter drop-down above the main area.

### Open findings overview

The **Open findings overview** panel displays the total number of open findings, as well as the number of findings of each severity.

To access the findings page with the corresponding filter applied, click on a number.

![Security and risk management open findings overview panel](images/security-risk-management-overview-open.png)<!-- TODO TAROT-2591: Screenshot -->

### Open finding distribution

The **Open finding distribution** panel displays the relative distribution of open findings by scan kind, severity, or status.

To select the desired distribution, use the drop-down in the top right-hand corner of the panel.

To access the findings page with the corresponding filter applied, click on a number.

![Security and risk management open finding distribution panel](images/security-risk-management-overview-distribution.png)<!-- TODO TAROT-2591: Screenshot -->

### Open finding history

> The **Open finding history** panel displays a graph showing weekly trends of open findings from the past three months, grouped by severity.

Reviewer 4: findings 💅🏻

Reviewer 2: Suggestion to avoid the same "The XXX panel displays" in all sections.
It would maybe deviate from the current **panel** usage, but we can use **area** instead, like we do on the [Organization overview](https://github.com/codacy/docs/blob/master/docs/organizations/organization-overview.md?plain=1#L24)... 🤔  

```suggestion
The **Open findings history** graph shows weekly trends of open findings from the past three months, grouped by severity.
```

For a detailed view of the distribution on a specific week, hover over the graph.

![Security and risk management open finding history panel](images/security-risk-management-overview-history-open.png)<!-- TODO TAROT-2591: Screenshot -->

### Activity history

> The **Activity history** panel displays a graph showing weekly counts of open and closed findings over the past three months and the overall open finding trend.

Reviewer 4: findings here again (?) 😅

Author: Yup, the idea is to mention that we have weekly counts (open/closed) plus the overall trend. I'll rephrase.

### Top 10 common security categories

The **Top 10 common security categories** panel displays the most common security categories of open findings, ordered by count.

To access the findings page with the corresponding filter applied, click on a category.

![Security and risk management top categories panel](images/security-risk-management-overview-top-categories.png)<!-- TODO TAROT-2591: Screenshot -->

## Findings {: id="item-list"}

The **Security and risk management findings** page displays a filtered list of findings, sorted by status. Use this page to review and prioritize findings and track the progress of your security efforts.

To access the overview page, select an organization from the top navigation bar, select **Security and risk** on the left navigation sidebar, and click the **Findings** tab in the top bar.

![Security and risk management findings page](images/security-risk-management-findings.png)<!-- TODO TAROT-2591: Screenshot -->

When viewing the findings, you can update the filtering criteria by clicking the **Severity**, **Status**, **Repository**, **Security category**, or **Scan type** drop-downs above the list.

> ### Viewing a single finding

Reviewer 2: Do we need a section here? 🤔

Author: We'll have a new single-finding detail view with some information that may be of use. It's more of a placeholder for now.

Author: I'll remove this for now; it will be implemented later.

To view the details of a single finding, click the **Details** column of the finding in the list.

<!-- TODO TAROT-2591: Draft section -->

![Security and risk management single finding page](images/security-risk-management-findings-single.png)<!-- TODO TAROT-2591: Screenshot -->

## Sharing a filtered view of findings {: id="sharing-filtered-view"}

To share the current view of the overview or findings page, click the **Copy URL** button in the top right-hand corner of the page. This action copies the URL with the current filters applied to the clipboard.

> 

Reviewer 2: Consider adding a screenshot of the screen area scoped to the button.

Author: I'll leave the screenshot out for now, since we're telling the user where to look for it. Open to reviewing later.

## Exporting findings {: id="exporting-the-security-item-list"}

!!! info "This feature is available only to organization admins and organization managers"

> To export a list of findings as a CSV file, click the options menu in the top right-hand corner of the page and select **Export CSV**. The exported list always includes all findings, ignoring any applied filters.

Reviewer 4: this option is renamed now 🌟 
![Screenshot 2024-03-14 at 18 00 04](https://github.com/codacy/docs/assets/28684353/0b5834b3-7940-40b2-943a-92080ea812e4)

Reviewer 2: Consider adding a screenshot of the menu.

Author: ```suggestion
To export a list of findings as a CSV file, click the options menu in the top right-hand corner of the page and select **Export findings (.csv)**. The exported list always includes all findings, ignoring any applied filters.
```

![Security and risk management activity history panel](images/security-risk-management-overview-history-activity.png)<!-- TODO TAROT-2591: Screenshot -->

### Top 10 high-risk repositories

The **Top 10 high-risk repositories** panel displays the repositories with the highest number of open findings, ordered by severity.

!!! note
    This panel may list less than ten repositories if there are fewer than ten repositories with open findings in the organization or if less than ten repositories are selected in the dropdown filter.

![Security and risk management high risk repositories panel](images/security-risk-management-overview-top-risk.png)<!-- TODO TAROT-2591: Screenshot -->

### Top 10 common security categories

The **Top 10 common security categories** panel displays the most common security categories of open findings, ordered by count.

To access the findings page with the corresponding filter applied, click on a category.

![Security and risk management top categories panel](images/security-risk-management-overview-top-categories.png)<!-- TODO TAROT-2591: Screenshot -->

## Findings {: id="item-list"}

> The **Security and risk management findings** page displays a filtered list of findings, sorted by status. Use this page to review and prioritize findings and track the progress of your security efforts.

Author: TODO confirm sorting options. What will be available at launch?

Reviewer 4: they will be available at a second stage. 
right now, we are going to deliver the first overview tab + findings tab with the table that exists on the items page. 
so i would hold this filtering part for now 🙇🏻‍♀️
